Sources/ArcGISToolkit/Components/Compass/Needle.swift

```diff
@@ -23,22 +23,51 @@ struct Needle: View {
                     HStack(spacing: 0) {
                         NeedleQuadrant(color: .lightRed)
                         NeedleQuadrant(color: .darkRed)
+#if os(visionOS)
+                            .perspectiveRotationEffect(
+                                .degrees(180),
+                                axis: (x: 0, y: 1, z: 0)
+                            )
+#else
                             .rotation3DEffect(
-                                Angle(degrees: 180),
-                                axis: (x: 0, y: 1, z: 0))
+                                .degrees(180),
+                                axis: (x: 0, y: 1, z: 0)
+                            )
+#endif
```

Comment:
Using `rotation3DEffect` was causing parts of the needle to flicker on a Vision Pro device not a simulator. And it turns out we shouldn't use this on visionOS according to [Apple](https://developer.apple.com/documentation/swiftui/view/rotation3deffect(_:axis:anchor:anchorz:perspective:)):
![Screenshot 2025-01-09 at 11 53 26 AM](https://github.com/user-attachments/assets/88cb4123-b9c8-4a65-a2b5-5e12c4b71282)

Using the method they recommend resolves this problem.